File: 2do_DAM/PMDM/1ra_Evaluacion/Ejemplos/Java/Ejemplo07_menuContexto/app/src/main/java/com/example/ejemplo_07_menucontexto/MainActivity.java
package com.example.ejemplo_07_menucontexto;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.ContextMenu;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.google.android.material.snackbar.Snackbar;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        registerForContextMenu(findViewById(R.id.lyPrincipal));
    }

    // Este metodo identifica el metodo para gestionar la posición del elemento seleccionado
    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {

//        if (item.getItemId()==R.id.menu_opcion1){
//            Log.d("Depurando","Opcion 1 pulsado");
//
//        }
            Snackbar.make(findViewById(R.id.lyPrincipal), item.getTitle().toString(), Snackbar.LENGTH_SHORT).show();
        return super.onOptionsItemSelected(item);
    }

    // Este metodo se encarga de visualizar el menu de aplicación crea el xml que le ha dado
    // del menu y visualizarlo cuando le des a los 3 puntitos
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu,menu);
        return true;
    }

    // Este metodo es para el menu contextual
    @Override
    public void onCreateContextMenu(ContextMenu menu, View v, ContextMenu.ContextMenuInfo menuInfo) {
        super.onCreateContextMenu(menu, v, menuInfo);
    }


    @Override
    public boolean onContextItemSelected(@NonNull MenuItem item) {
        return super.onContextItemSelected(item);
    }
}

File: 2do_DAM/PMDM/1ra_Evaluacion/Ejemplos/Kotlin/Ejemplo021_Toolbar/app/src/main/java/com/example/ejemplo021_toolbar/MainActivity.kt
package com.example.ejemplo021_toolbar

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import com.example.ejemplo021_toolbar.databinding.ActivityMainBinding
import com.google.android.material.snackbar.Snackbar

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setSupportActionBar(binding.mytoolbar.toolbar)
        binding.bSegundoActivity.setOnClickListener {
            startActivity(Intent(this, MainActivity2::class.java))
        }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.item1 -> {
                Snackbar.make(binding.root, "Item 1", Snackbar.LENGTH_SHORT).show()
            }

            R.id.item2 -> {
                Snackbar.make(binding.root, "Item 2", Snackbar.LENGTH_SHORT).show()
            }
        }
        return super.onOptionsItemSelected(item)
    }
}